storage/rocksdb/ha_rocksdb.cc

```diff
@@ -4331,8 +4356,26 @@ void ha_rocksdb::load_auto_incr_value() {
   const bool save_keyread_only = m_keyread_only;
   m_keyread_only = true;
 
-  if (!index_last(table->record[0]))
```

Comment:
I think you still need this codepath to handle upgrades correctly. There should be a test case for upgrade a server containing a table that has several rows from a build without the auto_increment data dictionary key to one with the auto_increment data dictionary key.

I.e.

```
/* load old build */
create table t1 (a int primary key auto increment)) engine=rocksdb;
insert into t1 values (NULL), (NULL);
/* shutdown, load new build */
insert into t1 values (NULL), (NULL);
```

You'll want to verify when the new build starts up, the inserts will use the correct auto_incr value, instead of resetting it to 0, which is what happens below.